src/nspawn/nspawn.c

```diff
@@ -6072,10 +6072,8 @@ static int run(int argc, char *argv[]) {
 
                         {
                                 BLOCK_SIGNALS(SIGINT);
-                                r = copy_file_full(arg_image, np, O_EXCL, arg_read_only ? 0400 : 0600,
-                                                   FS_NOCOW_FL, FS_NOCOW_FL,
-                                                   COPY_REFLINK|COPY_CRTIME|COPY_SIGINT,
-                                                   NULL, NULL);
+                                r = copy_file(arg_image, np, O_EXCL, arg_read_only ? 0400 : 0600,
+                                              COPY_REFLINK|COPY_CRTIME|COPY_SIGINT);
```

Comment:
here too. I think we shoud try to set the nocow flag here unconditionally on the copy. It has no ill effec afaics (not much positive effect either, but still tells the indication clearly to the fs)

i see the problem that reflinks cannot work if nocow is on. but that's fine, I think that just means fd_copy_regular() should be more careful here, and if COPY_REFLINK is set in conjunction with a request for FS_NOCOW_FL, it should do the copying first, and just nocow after.